UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,460 @@
+```go
+app.ProtocolPoolKeeper = protocolpoolkeeper.NewKeeper(
+		appCodec,
+		runtime.NewKVStoreService(keys[protocolpooltypes.StoreKey]),
+		app.AccountKeeper,
+		app.BankKeeper,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+)
+```
+
+Pass the protocolpool keeper to the distribution keeper:
+
+```go
+app.DistrKeeper = distrkeeper.NewKeeper(
+		appCodec,
+		runtime.NewKVStoreService(keys[distrtypes.StoreKey]),
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.StakingKeeper,
+		authtypes.FeeCollectorName,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		distrkeeper.WithExternalCommunityPool(app.ProtocolPoolKeeper), // NEW
+)
+```
+
+Add the protocolpool module to the module manager:
+
+```go
+app.ModuleManager = module.NewManager(
+		// ...
+		protocolpool.NewAppModule(appCodec, app.ProtocolPoolKeeper, app.AccountKeeper, app.BankKeeper),
+)
+```
+
+Add an entry for SetOrderBeginBlockers, SetOrderEndBlockers, SetOrderInitGenesis, and SetOrderExportGenesis.
+
+```go
+app.ModuleManager.SetOrderBeginBlockers(
+		// must come AFTER distribution.
+        distrtypes.ModuleName,
+		protocolpooltypes.ModuleName,
+)
+```
+
+```go
+app.ModuleManager.SetOrderEndBlockers(
+		// order does not matter.
+		protocolpooltypes.ModuleName,
+)
+```
+
+```go
+app.ModuleManager.SetOrderInitGenesis(
+	// order does not matter.
+    protocolpooltypes.ModuleName,   
+)
+```
+
+```go
+app.ModuleManager.SetOrderInitGenesis(
+    protocolpooltypes.ModuleName, // must be exported before bank.
+    banktypes.ModuleName,
+)
+```
+
+### DI Wiring
+
+First, set up the keeper for the application.
+
+Import the protocolpool keeper:
+
+```go
+	protocolpoolkeeper "github.com/cosmos/cosmos-sdk/x/protocolpool/keeper"
+```
+
+Add the keeper to your application struct:
+
+```go
+ProtocolPoolKeeper protocolpoolkeeper.Keeper
+```
+
+Add the keeper to the depinject system:
+
+```go
+depinject.Inject(
+	    appConfig,
+		&appBuilder,
+		&app.appCodec,
+		&app.legacyAmino,
+		&app.txConfig,
+		&app.interfaceRegistry,
+		// ... other modules
+		&app.ProtocolPoolKeeper, // NEW MODULE!
+	)
+```
+
+Next, set up configuration for the module.
+
+Import the following:
+
+```go
+import (
+    protocolpoolmodulev1 "cosmossdk.io/api/cosmos/protocolpool/module/v1"
+	
+    _ "github.com/cosmos/cosmos-sdk/x/protocolpool" // import for side-effects
+    protocolpooltypes "github.com/cosmos/cosmos-sdk/x/protocolpool/types"
+)
+```
+
+The protocolpool module has module accounts that handle funds. Add them to the module account permission configuration:
+
+```go
+moduleAccPerms = []*authmodulev1.ModuleAccountPermission{
+    // ...
+    {Account: protocolpooltypes.ModuleName},
+    {Account: protocolpooltypes.ProtocolPoolEscrowAccount},
+}
+```
+
+Next, add an entry for BeginBlockers, EndBlockers, InitGenesis, and ExportGenesis.
+
+```go
+BeginBlockers: []string{
+        // ...
+	    // must be AFTER distribution.
+		distrtypes.ModuleName,
+        protocolpooltypes.ModuleName,
```

Comment:
formatting weirdness here and a few other small places

Reply:
should be fixed